### ISSUE TITLE
HEAD requests with Node v5 have issues with gzipped content and fail.

### ISSUE THREAD

**Issue by czardoz** [request/request#2045]
This script reproduces the issue:

````js    
    var req = require('request');
    
    req({
        url: 'http://google.com',
        method: 'HEAD',
        headers: {},
        followAllRedirects: false,
        followRedirect: true,
        jar: true,
        timeout: 15000,
        gzip: true,
        rejectUnauthorized: false,
        strictSSL: true
    }, function (err, res, body) {
        if (err) {
            console.error(err.stack || err);
        }
    });
```

Steps to reproduce:

    aniket [/tmp/request-bug] -> node -v
    v5.1.0
    aniket [/tmp/request-bug] -> npm -v
    3.3.12
    aniket [/tmp/request-bug] -> cat node_modules/request/package.json| grep version
      "version": "2.69.0"
    aniket [/tmp/request-bug] -> node reproduce.js 
    Error: unexpected end of file
        at Zlib._handle.onerror (zlib.js:363:17)
    aniket [/tmp/request-bug] ->

**Comment by nathanmarks** [comment]
@czardoz FWIW I had the same issue with some HEAD requests and it is because of `followAllRedirects`. Unless you set it to `true`, it won't follow HEAD redirects (IE non www to www, or http to https).

**Comment by czardoz** [comment]
Is this issue reproducible for others? 

@simov Is there a label for a "confirmed" bug? I can consistently reproduce it with Node v5.

**Comment by nathanmarks** [comment]
@czardoz Not sure if this helps you, but I was doing `HEAD` requests to avoid downloading entire responses.

I ended up switching to `GET` requests, streaming the response and aborting the request early.

**Comment by czardoz** [comment]
@nathanmarks While that's a viable workaround, I use this library to run user-specified HTTP requests, so I cannot really change the request method :)

**Comment by simov** [comment]
Fixed here #2176

Let me know what do you think.
